core/templates/pages/mock-ajs.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Mock angular object for pages that don't use AngularJS.
+ */
+
+/**
+ * We are at a stage where we can have some pages in Angular. But we share
+ * component in between pages. So we can't remove the code that downgrades
+ * components and services. This code is found in most Angular components and
+ * services (towards the end of the file). In order to not have to create a
+ * separate file for each service and component just for the sake of downgrading
+ * , we have created this mock AngularJS object that will be used in the pages
+ * that don't use AngularJS.
+ */
+
+// TODO(#13080): Remove the mock-ajs.ts file after the migration is complete.
+
+import { VERSION } from '@angular/core';
+
+let mockAngular = {
+  component: () => mockAngular,
+  config: () => mockAngular,
+  constant: () => mockAngular,
+  controller: () => mockAngular,
+  directive: () => mockAngular,
+  factory: () => mockAngular,
+  filter: () => mockAngular,
+  module: () => mockAngular,
+  requires: () => [],
+  run: () => mockAngular,
+  service: () => mockAngular,
+  value: () => mockAngular,
+  version: VERSION
```

Comment:
**Why version of angular is added to a mock object of angularjs?**
Due to the presence of the `window.angular` object, protractor was considering this page as an `angularjs` page.
So, check on line 732 passes and definitelyNg1 is set to true.

But then control flow goes to line 740, but the if condition evaluates to `false` as resumeBootstrap function is not available in `angular 2 or greater`. As `defnitelyNg2OrNewer` is also false, none of callbacks are getting returned from this function.
As a result, the protractor keeps waiting, and the suite times out with an error.

(https://github.com/angular/protractor/blob/master/lib/clientsidescripts.js)
![image](https://user-images.githubusercontent.com/24855641/122073426-085f5580-ce16-11eb-99b0-4712d2e33f9d.png)

Fix:
Our aim was `definitelyNg2OrNewer` to be true and `definitelyNg1` to be false.
As we needed window.angular object but also require the condition on 732 to be false.
So, we add a version field of `window.angular` object having `value > 1`.

Another fix would be to define resumeBootstrap in the mock angular object but we find it more suitable if the protractor does not think of about page as angular page.